### PR TITLE
perf(spec-parser): support api key auth when add new api

### PR DESCRIPTION
**Pull request by SLdragon**

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -420,6 +420,7 @@
   "core.summary.actionSucceeded": "%s was executed successfully.",
   "core.summary.createdEnvFile": "Environment file was created at",
   "core.copilot.addAPI.success": "%s have(has) been successfully added to %s",
+  "core.copilot.addAPI.InjectAPIKeyActionFailed": "Inject API key action to teamsapp yaml file failed, please make sure that teamsapp yaml file contains teamsApp/create action in provision section.",
```
Comment (yuqizhou77) [inline packages/fx-core/resource/package.nls.json:423]: teamsApp.yml?

**Review by yuqizhou77 — DISMISSED** [review]

File: packages/fx-core/src/core/FxCore.ts [hunk 2]
```diff
@@ -1143,16 +1150,65 @@ export class FxCore {
...
+          const item = provisionNode.items[i];
+          if (item.get("uses") === "teamsApp/create") {
+            const teamsAppId = item.get("writeToEnvironmentFile")?.get("teamsAppId") as string;
+            if (teamsAppId) {
+              provisionNode.items.splice(i + 1, 0, {
+                uses: "apiKey/create",
+                with: {
+                  name: `${authName}`,
+                  appId: `\${{${teamsAppId}}}`,
+                  apiSpecPath: "./apiSpecificationFiles/openapi.yaml",
```
Comment (nliu-ms) [inline packages/fx-core/src/core/FxCore.ts:1181]: should we read the path of apiSpecFile from manifest.json?

**Review by nliu-ms — APPROVED** [review]

File: packages/fx-core/src/core/FxCore.ts (hunk 2 above)
Comment (SLdragon) [reply packages/fx-core/src/core/FxCore.ts:1181]: OK, thanks~

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (sffamily) [reply packages/fx-core/resource/package.nls.json:423]: agree

**Review by sffamily — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (SLdragon) [reply packages/fx-core/resource/package.nls.json]: udpated

**Review by nliu-ms — APPROVED** [review]

**Review by yuqizhou77 — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

File: packages/fx-core/src/core/FxCore.ts [hunk 3]
```diff
@@ -1143,16 +1150,69 @@ export class FxCore {
...
+  ): Promise<void> {
+    const ymlContent = await fs.readFile(ymlPath, "utf-8");
+
+    const document = parseDocument(ymlContent);
+    const provisionNode = document.get("provision") as any;
+
+    if (provisionNode) {
+      const hasApiKeyAction = provisionNode.items.some(
+        (item: any) =>
+          item.get("uses") === "apiKey/create" && item.get("with")?.get("name") === authName
```
Comment (adashen) [inline packages/fx-core/src/core/FxCore.ts:1167]: do we have central place for action names?

Comment (SLdragon) [reply packages/fx-core/src/core/FxCore.ts:1167]: No, I don't find central place of these names, each action hold its own action name.

**Review by adashen — APPROVED** [review]

**Review by sffamily — APPROVED** [review]

**Pull request merged**